client/src/components/DragAndDrop/drag&drop.jsx

```diff
@@ -0,0 +1,91 @@
+import { reorderColumn, reorderColumnCardMap } from "./reorder";
+import { DragDropContext, Droppable } from "react-beautiful-dnd";
+import BoardColumn from './column'
+import dndStyles from './assets/dndStyles'
+import { Box, Grid, Typography } from '@material-ui/core';
+import { useBoardContext } from '../../context/useBoardContext';
+
+
+
+const Board = () => {
+    const classes = dndStyles();
+    const { boardColumnMap: columns, board, ordered, setOrdered, setBoardColumnMap, boardActions, boardActionsInit } = useBoardContext();
+
+    const onDragEnd = result => {
+        const { type, combine, source, destination, droppableId, index} = result
+        
+        if (combine) {
+            if (type === "COLUMN") {
+                const shallow = [...ordered].splice(source.index, 1);
+                setOrdered(shallow);
+                return;
+            }
+
+            const column = columns[source.droppableId];
+            const withItemRemoved = [...column].splice(source.index, 1);
+            setBoardColumnMap({columns, [source.droppableId]: withItemRemoved})
+            return;
+        }
+
+        if (!destination) return;
+        
+        if (droppableId === destination.droppableId &&
+            index === destination.index) return;
+
+        if (type === "COLUMN") {
+            const $_ordered = reorderColumn(ordered, source.index, destination.index);
+            setOrdered($_ordered);
+            return;
+        }
+
+        const updatedData = reorderColumnCardMap({
+            columnMap: columns,
+            source,
+            destination
+        });
+        setBoardColumnMap(updatedData.columnMap)
+    };
+
+
+    return (<Grid className={classes.boardContainer} container item xs={12} sm={12}>
+        <Grid item xs={12} sm={12} className={classes.boardTitle}>
+            <Typography variant='h2'>{board.name}</Typography>
+        </Grid>
+
+        <Grid container item xs={12} sm={12}>
+            <Grid className={classes.boardAside} item xs={12} sm={1}>
+                <Typography>+</Typography>
+            </Grid>
+            
+            <Grid item xs={12} sm={10}>
+            <DragDropContext onDragEnd={onDragEnd}>
+                    <Droppable droppableId="board" type="COLUMN" direction="horizontal">
+                        {provided => (
+                        <Grid container item xs={12} sm={12} className={classes.columnParentContainer}  spacing={2} sx={{marginTop:'50px'}} ref={provided.innerRef} {...provided.droppableProps}>
+                                {ordered.map((key, index) => (
+                                    <Box key={key} className={classes.columnContainer} >
```

Comment:
Can we name these more descriptively - for example, as a collaborator I can't know what this code does without also observing the `ordered` variable in the context. Does it follow a format such as `columnName: columnIndex`? Or is it `someSortOfColumnId: columnOrder`? Hope that makes sense 😇 

In general, I would recommend staying away from vague variable names such as `key`, `index`, `value`, `data`...